templates/rc.erb

```diff
@@ -0,0 +1,263 @@
+#! /bin/sh
+#
+# rc
+#
+# Starts/stops services on runlevel changes.
+#
+# Optimization: A start script is not run when the service was already
+# configured to run in the previous runlevel.  A stop script is not run
+# when the the service was already configured not to run in the previous
+# runlevel.
+#
+# Authors:
+# 	Miquel van Smoorenburg <miquels@cistron.nl>
+# 	Bruce Perens <Bruce@Pixar.com>
```

Comment:
> Again.. As I am super new to puppet and hardening, just started last month.

@bitvijays I understand and see that, do not get my questions wrong, I do not want to punish or whatever, I'm only supporting you and project by review and questions :-) If you get an impression my feedback is harsh, thats not my intention, please give me feedback on this and I'll try to improve my communication in the future ;)

> which I took it from my debian system. 

Yes, this is basically a problem at this specific point. If you take foreign code, you have to take care about the licensing question. It means: either to check and prove the license compatibility and terms of distribution/relicensing or to get an approval from all authors of code for relicensing/redistribution. (see https://github.com/fraenki/puppet-sshkeys/issues/11 as example where I was asked a similar question)

This is a very important topic in the OpenSource field, nobody wants to have legal claims/discussions because of license violations.